m2cgen/assemblers/svm.py

```diff
@@ -135,3 +170,39 @@ def _linear_kernel_with_gama_and_coef(self, support_vector):
         kernel = self._linear_kernel(support_vector)
         kernel = utils.mul(self._gamma_expr, kernel)
         return utils.add(kernel, ast.NumVal(self.model.coef0))
+
+
+class LightningSVMModelAssembler(SklearnSVMModelAssembler):
+
+    def _get_supported_kernels(self):
+        kernels = super()._get_supported_kernels()
+        kernels["cosine"] = self._cosine_kernel
+        return kernels
+
+    def _get_gamma(self):
+        return self.model.gamma
+
+    def _get_output_size(self):
+        return 1
+
+    def _assemble_multi_class_output(self):
+        raise NotImplementedError
+
+    def _get_single_coef(self):
+        return self.model.coef_[0]
+
+    def _cosine_kernel(self, support_vector):
+        support_vector_norm = np.linalg.norm(support_vector)
+        feature_norm = ast.SqrtExpr(
+            utils.apply_op_to_expressions(
+                ast.BinNumOpType.ADD,
+                *[utils.mul(ast.FeatureRef(i), ast.FeatureRef(i))
+                  for i in range(len(support_vector))]),
+            to_reuse=True)
```

Comment:
Have no idea how to avoid zero norm 🙁

Reply:
What is the lighting's behavior for zero future norm? Can we get away with the same logic that we use for Support Vector norm?

Reply:
It relies on scikit-learn code:
https://github.com/scikit-learn/scikit-learn/blob/95d4f0841d57e8b5f6b2a570312e9d832e69debc/sklearn/preprocessing/_data.py#L1730
https://github.com/scikit-learn/scikit-learn/blob/95d4f0841d57e8b5f6b2a570312e9d832e69debc/sklearn/preprocessing/_data.py#L63-L78

Reply:
If I understand the logic correctly it does the same thing (falls back to 1.0), shall we do the same here?

Reply:
To be honest, I'm not sure that complicating generated code in a such way worth it...

Reply:
I'd say we should always prefer correctness even if it means a slightly more complex code. Btw, does it really become that complex? Especially considering that we reuse the expression (to_reuse=True) here. AST doesn't look bad at all. Thanks for implementing this! 👍

Reply:
> we should always prefer correctness even if it means a slightly more complex code. 

Yeah, no doubt!

In languages which implement `to_reuse=True` it looks OK, but if they don't... Though, there are many other places where without caching code looks terrible.

I thought about more elegant solution, but didn't come to any.

Reply:
> I thought about more elegant solution, but didn't come to any.
Well, TBH, the existing solution looks more than good to me at least at this point 👍